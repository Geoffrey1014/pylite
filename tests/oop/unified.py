class Vector(list):
    def __init__(self):
        list.__init__(self)
        self.count = 0

    def push_back(self, item):
        self.append(item)
        self.count += 1

vector = Vector()
vector.push_back(1)
vector.push_back(2)
vector.push_back(3)

print 'vector.count = %d' % vector.count
for item in vector:
    print item
