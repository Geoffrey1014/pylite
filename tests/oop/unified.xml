lambda
  name __init__
  param_count 0
  local_count 1
  label_count 3
  insn_vector
    new_class c0
    store_global c0
    load_global c1
    load_global c0
    inherit
    load_const c2
    load_global c0
    store_field c3
    load_const c4
    load_global c0
    store_field c5
    load_global c0
    call 0
    store_global c6
    load_const c7
    load_global c6
    load_field c5
    call 1
    pop
    load_const c8
    load_global c6
    load_field c5
    call 1
    pop
    load_const c9
    load_global c6
    load_field c5
    call 1
    pop
    load_global c6
    load_field c10
    load_const c11
    mod
    print
    load_global c6
    iterator
    store_local -1
    jmp L1
    label L0
    store_global c12
    load_global c12
    print
    label L1
    load_local -1
    next L0
    pop
    label L2
    load_const c13
    return
  const_vector
    Vector
    list
    lambda
      name __init__
      param_count 1
      local_count 0
      label_count 0
      insn_vector
        load_local 0
        load_global c0
        load_field c1
        call 1
        pop
        load_const c2
        load_local 0
        store_field c3
        load_const c4
        return
      const_vector
        list
        __init__
        0.000000
        count
        None
      guard_vector
      line_number_map 3 3 3 3 3 4 4 4 2 2
    __init__
    lambda
      name push_back
      param_count 2
      local_count 0
      label_count 0
      insn_vector
        load_local 1
        load_local 0
        load_field c0
        call 1
        pop
        load_const c1
        load_local 0
        load_field c2
        add
        load_local 0
        store_field c2
        load_const c3
        return
      const_vector
        append
        1.000000
        count
        None
      guard_vector
      line_number_map 7 7 7 7 7 8 8 8 8 8 8 6 6
    push_back
    vector
    1.000000
    2.000000
    3.000000
    count
    "vector.count = %d"
    item
    None
  guard_vector
  line_number_map 1 1 1 1 1 2 2 2 6 6 6 10 10 10 11 11 11 11 11 12 12 12 12 12 13 13 13 13 13 15 15 15 15 15 16 16 16 16 16 16 17 17 16 16 16 16 16 1 1
