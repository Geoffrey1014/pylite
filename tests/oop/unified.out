vector.count = 3
1
2
3
